core/flyout.js

```diff
@@ -446,7 +605,13 @@ Blockly.Flyout.prototype.show = function(xmlList) {
  * @private
  */
 Blockly.Flyout.prototype.layoutBlocks_ = function(blocks, gaps, margin) {
-  // Lay out the blocks vertically.
+  // The blocks need to be visible in order to be laid out and measured
+  // correctly, but we don't want the flyout to show up until it's properly
+  // sized.  Opacity is reset at the end of position().
+  this.svgGroup_.style.opacity = 0;
```

Comment:
In the vertical case we didn't care about how much space the blocks took up until after reflow.  We cared about how much space they took up for setting the background path and also when we measured for scrolling/resizing.  In the horizontal rtl case we need to measure them correctly on the first pass so we can offset them such that the last block is near (0, 0).